### DISCUSSION TITLE
Specify 'logs root' property of L2 Blocks

### DISCUSSION THREAD

**Discussion by maurelian** [ethereum-optimism/optimistic-specs#228] [Design Decisions]
> We should consider including a "logs root" (something more neat than the receipt root on L1). 
> Merkleizing the L1 log events more efficiently, filtered to only those of the successful receipts, would be a nice thing to do as rollup. If L2 doesn't pay for a log-subscription in advance (the extension we talked about), at least it can efficiently provide a proof later on that something happened on L1.

This issue is a placeholder, taken from a [PR comment here](https://github.com/ethereum-optimism/optimistic-specs/pull/43/files#r759672626) by @protolambda.

**Comment by norswap** [comment]
Is this worth it? The merkleized entries in the receipts root comprise a concatenation of all log events emitted during a transaction. Seems like except extremely weird cases, this should not be too big? [EIP-4488](https://github.com/ethereum/EIPs/pull/4488) will also drastically reduce the cost of proving the entry.

**Comment by protolambda** [comment]
It's not about the cost of the fraud-proof here, but about the cost for a L2 user inside the L2 EVM to casually proof some L1 event.

If we did the heavy lifting by providing a nice to work with accumulator then L2 users will have to pay less fees and it may also be less error prone for them (a lot easier to verify a plain byte array of a log entry in a binary tree, than to verify a RLP-encoded receipt in a MPT)
